### PR TITLE
[c/en] mostly word clarification and extra descriptions

### PR DESCRIPTION
**Pull request by awschult002**
- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]` (example `[python/fr-fr]` or `[java/en]`)
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [x] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

### CODE REVIEW

File: c.html.markdown [hunk 1]
```diff
@@ -46,31 +46,47 @@ Multi-line comments don't nest /* Be careful */  // comment ends on this line...
...
 
 // Declare function signatures in advance in a .h file, or at the top of
 // your .c file.
 void function_1();
 int function_2(void);
 
-// Must declare a 'function prototype' before main() when functions occur after
-// your main() function.
+// At a minimum, you must declare a 'function prototype' before its use in any function.
+// Normally, protoypes are placed at the top of a file before any function definition.
```
Comment (Menelion) [inline c.html.markdown:75]:
```suggestion
// Normally, prototypes are placed at the top of a file before any function definition.
```

File: c.html.markdown [hunk 2]
```diff
@@ -46,31 +46,47 @@ Multi-line comments don't nest /* Be careful */  // comment ends on this line...
...
 int add_two_ints(int x1, int x2); // function prototype
 // although `int add_two_ints(int, int);` is also valid (no need to name the args),
 // it is recommended to name arguments in the prototype as well for easier inspection
 
-// Your program's entry point is a function called
-// main with an integer return type.
+// Function protoypes are not necessary if the function definition comes before
```
Comment (Menelion) [inline c.html.markdown:80]:
```suggestion
// Function prototypes are not necessary if the function definition comes before
```

File: c.html.markdown [hunk 3]
```diff
@@ -46,31 +46,47 @@ Multi-line comments don't nest /* Be careful */  // comment ends on this line...
...
+// any other function that calls that function. However, it's standard practice to
+// always add the function prototype to a header file (*.h) and then #include that
+// file at the top. This prevents any issues where a function might be called
+// before the compiler knows of its existence; while also giving the developer a
```
Comment (Menelion) [inline c.html.markdown:84]:
```suggestion
// before the compiler knows of its existence, while also giving the developer a
```

File: c.html.markdown [hunk 4]
```diff
@@ -46,31 +46,47 @@ Multi-line comments don't nest /* Be careful */  // comment ends on this line...
...
+// clean header file to share with the rest of the project.
+
+// Your program's entry point is a function called "main". The return type can
+// be anything, however most operating systems expect a return type of 'int' for
```
Comment (Menelion) [inline c.html.markdown:88]:
```suggestion
// be anything, however most operating systems expect a return type of `int` for
```

File: c.html.markdown [hunk 5]
```diff
@@ -96,13 +112,14 @@ int main (int argc, char** argv)
   // For the sake of the tutorial, variables are declared dynamically under
   // C99-compliant standards.
 
-  // ints are usually 4 bytes
+  // ints are usually 4 bytes (use the 'sizeof' operator to check)
```
Comment (Menelion) [inline c.html.markdown:115]:
```suggestion
  // ints are usually 4 bytes (use the `sizeof` operator to check)
```

File: c.html.markdown [hunk 6]
```diff
@@ -96,13 +112,14 @@ int main (int argc, char** argv)
...
   int x_int = 0;
 
-  // shorts are usually 2 bytes
+  // shorts are usually 2 bytes (use the 'sizeof' operator to check)
```
Comment (Menelion) [inline c.html.markdown:118]:
```suggestion
  // shorts are usually 2 bytes (use the `sizeof` operator to check)
```

File: c.html.markdown [hunk 7]
```diff
@@ -96,13 +112,14 @@ int main (int argc, char** argv)
...
   short x_short = 0;
 
-  // chars are guaranteed to be 1 byte
+  // chars are defined as the smallest addressable unit for a processor. 
+  // This is usually 1 byte, but some systems are more (ex. TMS320 from TI is 2 bytes).
```
Comment (Menelion) [inline c.html.markdown:122]:
```suggestion
  // This is usually 1 byte, but for some systems it can be more (ex. for TMS320 from TI it's 2 bytes).
```

**Review by Menelion — CHANGES_REQUESTED** [review]
Thank you! Please fix those small typos and I'll merge.

File: c.html.markdown (hunk 5 above)
Comment (awschult002) [reply c.html.markdown:115]: i didn't realize that markdown would format code fences... inside of a code fence.

Comment (Menelion) [reply c.html.markdown]: I actually don't know it for sure, but it's easier to discern the code that way, on my opinion :)

**Review by Menelion — APPROVED** [review]
Thank you!

**Pull request merged**